### ISSUE TITLE
maven-deploy.sh should have same description in --repository flag as maven-deploy.bat

### ISSUE THREAD

**Issue by perlun** [jOOQ/jOOQ#12939]
Not being very experienced with Maven repositories, this puzzled me for a while, until I stumbled into the `maven-deploy.bat` file which had it more clearly described. I think it could be worth adding this to the `maven-deploy.sh` version of the script as well:

```diff
--- maven-deploy.org	2022-01-25 12:53:34.657838607 +0200
+++ maven-deploy.sh	2022-01-25 12:53:55.711763941 +0200
@@ -20,7 +20,7 @@
   echo ""
   echo "Options:"
   echo "  -h, --help                      Display this help and exit"
-  echo "  -r, --repository   (Optional)   The repository id"
+  echo "  -r, --repository   (Optional)   The repository id as specified in your settings.xml"
   echo "  -u, --url          (Mandatory)  The repository URL"
   exit 0
   ;;
```

**Comment by perlun** [comment]
On another note; would it be worth adding `set -e` (and potentially `-o pipefail`) to this script as well? Some people hate this kind of error handling (and it surely has some drawbacks), but... as the script currently stands, if the upload fails with e.g. `401 Unauthorized` or `403 Forbidden`, the script will happily carry on and try to upload all the other files as well.

I personally like the idea of "fail fast" and I think it could be applied here as well. Failing to upload one of the files could just as well error with some reasonable error message instead of just continuing with the rest of the files.

**Comment by lukaseder** [comment]
Thanks for your feature request. I agree with the improved help text. Maybe I'll add "e.g." because there are multiple ways to identify a repository location in Maven.

> On another note

Can you create a new issue? One issue per topic helps keep track of things, as well as produce clean release notes...

**Comment by lukaseder** [comment]
> until I stumbled into the `maven-deploy.bat` file which had it more clearly described

Ah, I just saw this phrase. Well, that's a bug, then! :)

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.3 (https://github.com/jOOQ/jOOQ/issues/12940), and 3.15.7 (https://github.com/jOOQ/jOOQ/issues/12941)

**Comment by perlun** [comment]
> Can you create a new issue? One issue per topic helps keep track of things, as well as produce clean release notes...

Yep, no problem at all: https://github.com/jOOQ/jOOQ/issues/12942
